File: lc_py/lcPy/array/findSmallestInteger.py
from typing import List
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        cnt = {}
        for x in nums:
            cnt[(x % value + value) % value] += 1
        mex = 0
        while cnt[mex % value]:
            cnt[mex % value] += 1
            mex += 1
        return mex


if __name__ == "__main__":
    nums = [1,-10,7,13,6,8]
    s = Solution()
    print(s.findSmallestInteger(nums, 5))  
    
            

File: lc_rust/src/bin/findSmallestInteger.rs
use std::collections::HashMap;

struct Solution;

impl Solution {
    pub fn find_smallest_integer(mut nums: Vec<i32>, m: i32) -> i32 {
        let mut cnt: HashMap<i32, usize> = HashMap::new();

        for x in nums {
            *cnt.entry((x % m + m) % m).or_insert(0) += 1;
        }
        for mex in 0.. {
            if let Some(c) = cnt.get_mut(&(mex % m)) {
                if *c > 0 {
                    *c -= 1;
                    continue
                }
            }
            return mex
        }
        unreachable!()
    }
}
fn main() {

}